unified-runtime/source/adapters/cuda/flags_provider.hpp

```diff
@@ -0,0 +1,109 @@
+
+#pragma once
+#include "umf_helpers.hpp"
+#include <umf/memory_pool.h>
+
+template <typename AllocMixin, typename FlagTy = unsigned, size_t ReserveSz = 4>
+class FlagsMemProviderWithDefaultBase {
```

Comment:
You can consider extending https://github.com/intel/llvm/blob/sycl/unified-runtime/source/common/ur_pool_manager.hpp with flags and using it here. However, it would require some changes in how pools are created in CUDA.

We will probably need to extend the pool manager to support flags anyway to enable this for L0.